search-sentiments-map/routes/trends.js

```diff
@@ -1,35 +1,233 @@
 // Copyright 2019 Google LLC
 //
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
+// Licensed under the Apache License, Version 2.0 (the 'License'); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
 //
 //     https://www.apache.org/licenses/LICENSE-2.0
 //
 // Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+// distributed under the License is distributed on an 'AS IS' BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
 
 /** Server-side script that gets data from the google-trends-api. */
 
 const express = require('express');
-var router = express.Router();  // Using Router to divide the app into modules.
+const router = express.Router();  // Using Router to divide the app into modules.
 
 const googleTrends = require('google-trends-api');
+const {Datastore} = require('@google-cloud/datastore');
+const datastore = new Datastore();
+const TRENDS_DATA_KIND = 'TrendsEntry';
 
-/** Render a JSON string of the top 20 (or fewer) search trends of the past 24 hours. */
+/** 
+ * Renders a JSON array of the top 20 (or fewer) global search trends maintained
+ * for the current 12-hour range.
+ */
 router.get('/', (req, res) => {
-  googleTrends.dailyTrends({
-    trendDate: new Date(),
-    geo: 'US',
-  }).then(dailyTrendsJson => {
+  retrieveGlobalTrends().then(globalTrends => {
     res.setHeader('Content-Type', 'application/json');
-    res.send(dailyTrendsJson);
-  }).catch(err => {
-    console.log(err);
+    res.send(globalTrends);
   });
 });
 
-module.exports = router;
+/** 
+ * Queries the Datastore for the most recent global trends.
```

Comment:
I would be a little more descriptive here since you're doing more than only "querying"

Reply:
It's just getting an entity but yeah query may not be the right word